src/App.test.jsx

```diff
@@ -0,0 +1,26 @@
+import { render } from '@testing-library/react';
+
+import { useDispatch, useSelector } from 'react-redux';
+
+import App from './App';
+
+jest.mock('react-redux');
+
+describe('App', () => {
+  const dispatch = jest.fn();
+
+  useDispatch.mockImplementation(() => dispatch);
+  useSelector.mockImplementation((selector) => selector({
+    locations: [],
+  }));
+
+  it('renders location list', () => {
+    const { queryByText } = render((
+      <App />
+    ));
+
+    expect(dispatch).toBeCalledTimes(1);
+
+    expect(queryByText('지역')).not.toBeNull();
```

Comment:
```suggestion
    const { container } = render((
      <App />
    ));

    expect(dispatch).toBeCalledTimes(1);

    expect(container).toHaveTextContent();
```

조건문이나 단정문에 부정문을 잘 이해하기 어려워서, 긍정문으로 변경할 수 있는 경우 긍정문을 사용하면 좋을 것 같습니다